app/assets/javascripts/admin/addon/components/schema-theme-setting/editor.gjs

```diff
@@ -226,19 +231,32 @@ export default class SchemaThemeSettingNewEditor extends Component {
     }
   }
 
+  get schemaModeKeys() {
+    if (this.schemaMode === SCHEMA_MODES.THEME) {
+      return {
+        id: this.args.themeId,
+        route: "adminCustomizeThemes.show",
+        schema: this.args.setting.objects_schema,
+      };
+    }
+    return {
+      id: this.args.pluginId,
+      route: "adminPlugins.show.settings",
+      schema: this.args.setting.schema,
+    };
+  }
```

Comment:
Instead of doing this, I think we can create a base component that `SchemaThemeSetting::Editor` and `SchemaPluginSetting::Editor` can inherit from?